Update Databricks naming (#13722)
airbytehq/airbyte: airbyte-integrations/connectors/destination-databricks/BOOTSTRAP.md

# Databricks Lakehouse Destination Connector Bootstrap

> The Databricks Lakehouse Connector enables a developer to sync data into a Databricks cluster. It does so in two steps:

Reviewer: Maria wants the doc to mention "Delta Lake on Databricks Lakehouse" in the first sentence.

```suggestion
This destination syncs data to Delta Lake on Databricks Lakehouse. It does so in two steps:
```
